plugins/clipboard-manager/Cargo.toml

```diff
@@ -25,3 +25,4 @@ thiserror = { workspace = true }
 
 [target."cfg(any(target_os = \"macos\", windows, target_os = \"linux\", target_os = \"dragonfly\", target_os = \"freebsd\", target_os = \"openbsd\", target_os = \"netbsd\"))".dependencies]
 arboard = "3"
+image   = "0.24"
```

Comment:
I am not too happy with the current state of images handling in tauri overall and before we merge this PR I think it would be much better to have a better representation in tauri core similar to electron's `NativeImage` class. 

cc @lucasfernog

Reply:
Also something i don't get about arboard, they already depend on the image crate but still created their own image type. imho super annoying...

Reply:
> isn't the image crate basically the ecosystem default for that? I'd much rather agree on one of its types (like RgbaImage or DynamicImage idk) across all of our crates than creating our own type(s) (again).

yes but image crate is a heavy dependency and you may not need it most of it, for that reason, crates like `winit`, `muda` and `tray-icon` opted into a light icon type.

What I meant by my previous comment is that we need something like `NativeImage` that is light like `winit`'s `Icon` type, with a few helper methods to support creating from:
- png / jpeg / ico bytes (based on feature flags enabled)
- png / jpeg / ico files (based on feature flags enabled)
- raw rbga8
- can have a lighter representation in JS (using tauri's new resource table) so we don't need to send bytes forth and back between JS and Rust (unless user wants to get it in JS)
- unlike the existing `tauri::Icon`, it will be a struct not an Enum

Reply:
Sounds like the helper functions would negate most of the "lightness" compared to image::RgbaImage honestly (with its default features disabled of course) but i don't have any hard feelings about it, as long as we have _one_ type we can re-use across all repos 🤷

Reply:
Not really, the helper functions will use the lighter crates `png` and `ico` and will be behind a feature flag as well.